### PR TITLE
corrects lineCompare, handles direction flag, covers MultiLineString

### PR DESCRIPTION
**Pull request by armantorkzaban**

### CODE REVIEW

File: lib/src/turf_equality_base.dart [hunk 1]
```diff
@@ -136,20 +135,26 @@ class Equality {
...
 
   bool _compareLine(LineString line1, LineString line2) {
-    for (var i = 0; i < line1.coordinates.length; i++) {
-      if (!_compareCoords(line1.coordinates[i], line2.coordinates[i])) {
-        if (direction) {
+    if (!_compareCoords(line1.coordinates.first, line2.coordinates.first)) {
+      if (directionMatters) {
+        return false;
+      } else {
+        var newLine = line2..coordinates = line2.coordinates.reversed.toList();
```
Comment (lukas-h) [inline lib/src/turf_equality_base.dart:142]: this is modifying the old line and not creating a new one!
The deleted code was correct

```dart
LineString(
              coordinates: line2.coordinates.reversed.toList(),
            ),
```

File: lib/src/turf_equality_base.dart [hunk 2]
```diff
@@ -81,6 +81,7 @@ class Equality {
     }
     //
     else if (_compareTypes<MultiPoint>(g1, g2)) {
+      print("reaches");
```
Comment (lukas-h) [inline lib/src/turf_equality_base.dart:84]: remove print in production

File: lib/src/turf_equality_base.dart [hunk 3]
```diff
@@ -182,20 +188,20 @@ class Equality {
...
               list1[i][positionIndex],
               (list2[i][(list2[i].length + positionIndex + diff) %
                   list2[i].length]))) {
             return false;
           }
-        } else if (direction && !shiftedPolygon) {
+        } else if (directionMatters && !shiftedPolygon) {
           List<List<Position>> listReversed = poly2
               .clone()
               .coordinates
```
Comment (lukas-h) [inline lib/src/turf_equality_base.dart:207]: can we write this a little bit more efficiently and cleaner? This way we would not have 4 `(else) if`-blocks but just two nested `if`-`else`s
like e.g.

```dart
if(directionMatters) {
    List<List<Position>> listReversed = poly2
              .clone()
              .coordinates
    // ...
    if(shiftedPolygon){
        // ...
    } else {
        // ...
    }
} else{
    if(shiftedPolygon){
        // ...
    } else{
        // ...
    }
}

**Review by lukas-h — CHANGES_REQUESTED** [review]
cloning mistake and refactoring

**Comment by armantorkzaban** [comment]
I go ahead and merge this. 
Do we release a new version with bug fixes?

**Pull request merged**